src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -77,6 +77,69 @@ internal static ParseResult ParseScript(string script, ILanguageSettings loc = n
             return new ParseResult(parsetree, errors, errors?.Any() ?? false, parser._comments, parser._before, parser._after);
         }
 
+        public static ParseFormulasResult ParseFormulasScript(string script, ILanguageSettings loc = null, Flags flags = Flags.None)
+        {
+            Contracts.AssertValue(script);
+            Contracts.AssertValueOrNull(loc);
+
+            Token[] formulaTokens = TokenizeScript(script, loc, flags);
+            TexlParser parser = new TexlParser(formulaTokens, flags);
+            List<TexlError> errors = null;
+
+            return parser.ParseFormulas(ref errors);
+        }
+
+        private ParseFormulasResult ParseFormulas(ref List<TexlError> errors)
+        {
+            Dictionary<DName, TexlNode> namedFormulas = new Dictionary<DName, TexlNode>();
+            while (_curs.TokCur.Kind != TokKind.Eof)
+            {
+                // Verify identifier
+                Token thisIdentifier = TokEat(TokKind.Ident);
+                if (thisIdentifier != null)
+                {
+                    // Verify "="
+                    Token thisEq = TokEat(TokKind.Equ);
+                    if (thisEq != null)
+                    {
+                        // Extract expression
+                        List<Token> expression = new List<Token>();
+                        while (_curs.TidCur != TokKind.Semicolon)
+                        {
+                            // Check if we're at EOF before a semicolon is found
+                            if (_curs.TidCur == TokKind.Eof)
+                            {
+                                CreateError(_curs.TokCur, TexlStrings.ErrNamedFormula_MissingSemicolon);
+                                return new ParseFormulasResult(namedFormulas, errors, errors?.Any() ?? false);
+                            }
+
+                            expression.Add(_curs.TokCur);
+                            _curs.TokMove();
+                        }
+
+                        // Parse expression
+                        
+                        TexlParser expressionParser = new TexlParser(expression.ToArray(),Flags.None);
```

Comment:
Can this whole thing be replaced with a call to ParseExpr() on the current instance?

Reply:
Done. I had to create a flag for the named formulas case.